### PR TITLE
feat(conventional-changelog-presets): supported new preset format

### LINKED ISSUE #525: promisify crashing node
I have several different packages where semantic release is no longer working.
The last line in the log is this:
```
[3:27:11 PM] [semantic-release] › ℹ  Start step "analyzeCommits" of plugin "@semantic-release/commit-analyzer"
```
The exit code is 0 so my pipelines proceed but no release is created.
This happens on several repositories, most open source, I'll link some CI traces here:
- https://github.com/collecthor/flysystem-adapters/actions/runs/6146191274/job/16675106648
- https://github.com/pheanstalk/pheanstalk/actions/runs/6123015944/job/16620078664

The code causing this issue lives in `load-parser-config.js`:

```js
export default async ({ preset, config, parserOpts, presetConfig }, { cwd, logger }) => {
  let loadedConfig;
  const __dirname = dirname(fileURLToPath(import.meta.url));

  if (preset) {
    const presetPackage = `conventional-changelog-${preset.toLowerCase()}`;
    loadedConfig = importFrom(cwd, presetPackage);
  } else if (config) {
    loadedConfig = importFrom.silent(__dirname, config) || importFrom(cwd, config);
  } else {
    loadedConfig = conventionalChangelogAngular;
  }

  // Comments added by @sammousa. 
  // loadedConfig is an async function, specifically the createPreset export from conventional-changelog-conventionalcommits package
  // which implements the conventional commit standard that this package uses
  loadedConfig = await (typeof loadedConfig === "function"
      ? isPlainObject(presetConfig)
          ? loadedConfig(presetConfig)
          // this line crashes nodejs silently when called with an async function
          : promisify(loadedConfig)()
      : loadedConfig);

  return { ...loadedConfig.parserOpts, ...parserOpts };
};

```

The bug is definitely in nodeJS, the fix for this package however would be to not use `promisify` at all, since we're calling the function directly if there IS a configuration, why not also call it directly when there isn't?

### PR DESCRIPTION
**Pull request by travi**
BREAKING CHANGE: the new preset format is a breaking change when compared to the previous preset format. updating to support the new format means that the old preset format is no longer supported. update your preset to the latest version to maintain compatibility

closes #525

### CODE REVIEW

**Comment by travi** [comment]
so far, this updates the conventional-changelog presets and adjusts loading to handle the new format. 

`conventional-commits-filter` has not yet been updated since it appears to not handle reverts the same way. if that is an expected change, i have not investigated far enough to understand what adjustment we need to make.

**Comment by travi** [comment]
got the integration test passing by reversing the list of commits to match the conventional-changelog expectations based on https://github.com/conventional-changelog/conventional-changelog/issues/1121#issuecomment-1709206347. however, assuming the previous order actually aligned with how core provides the list, this will need to be coordinated with reversing that list somewhere

File: lib/load-parser-config.js [hunk 1]
```diff
@@ -22,18 +22,12 @@ export default async ({ preset, config, parserOpts, presetConfig }, { cwd }) =>
...
-    loadedConfig = conventionalChangelogAngular;
+    loadedConfig = await conventionalChangelogAngular();
   }
 
-  loadedConfig = await (typeof loadedConfig === "function"
-    ? isPlainObject(presetConfig)
-      ? loadedConfig(presetConfig)
-      : promisify(loadedConfig)()
-    : loadedConfig);
-
```
Comment (travi) [inline lib/load-parser-config.js:37]: see additional explanation at https://github.com/semantic-release/release-notes-generator/pull/526#discussion_r1313604113

File: lib/load-parser-config.js [hunk 2]
```diff
@@ -22,18 +22,12 @@ export default async ({ preset, config, parserOpts, presetConfig }, { cwd }) =>
 
   if (preset) {
     const presetPackage = `conventional-changelog-${preset.toLowerCase()}`;
-    loadedConfig = importFrom.silent(__dirname, presetPackage) || importFrom(cwd, presetPackage);
+    loadedConfig = await (importFrom.silent(__dirname, presetPackage) || importFrom(cwd, presetPackage))(presetConfig);
   } else if (config) {
-    loadedConfig = importFrom.silent(__dirname, config) || importFrom(cwd, config);
+    loadedConfig = await (importFrom.silent(__dirname, config) || importFrom(cwd, config))();
```
Comment (travi) [inline lib/load-parser-config.js:27]: same as https://github.com/semantic-release/release-notes-generator/pull/526#discussion_r1313605497

**Review by gr2m — APPROVED** [review]

**Pull request merged**